hs_access_control/models.py

```diff
@@ -1273,62 +1274,97 @@ def get_resources_with_explicit_access(self, this_privilege, via_user=True, via_
+            else:
+                return BaseResource.objects\
+                    .filter(query).distinct()
 
-        else:  # this_privilege == PrivilegeCodes.ViEW
+        else:  # this_privilege == PrivilegeCodes.VIEW
             # VIEW includes CHANGE+immutable as well as explicit VIEW
             # TODO: make this query more efficient, if possible!
             # CHANGE does not include immutable resources
+
+            # step one: compute view resources explicitly:
             uquery = Q(r2urp__privilege=this_privilege,
                        r2urp__user=self.user)
             uexcl = Q(r2urp__privilege__lt=this_privilege,
                       r2urp__user=self.user)
-
             gquery = Q(r2grp__privilege=this_privilege,
                        r2grp__group__g2ugp__user=self.user)
             gexcl = Q(r2grp__privilege__lt=this_privilege,
                       r2grp__group__g2ugp__user=self.user)
 
             if via_user and via_group:
                 query = uquery | gquery
-                excl = uexcl | gexcl
+                excl = uexcl | gexcl  # user privilege can override group privilege
             elif via_user:
                 query = uquery
-                excl = uexcl
+                excl = None  # only one privilege per user
             elif via_group:
                 query = gquery
-                excl = gexcl
+                excl = gexcl  # CHANGE from other source squashes VIEW
             else:
                 # nothing matches
-                return BaseResource.objects.empty()
-
-            view = BaseResource.objects\
-                .filter(query)\
-                .exclude(pk__in=BaseResource.objects
-                         .filter(excl))
-
-            immutable = BaseResource.objects\
-                .filter(raccess__immutable=True,
-                        r2urp__privilege=PrivilegeCodes.CHANGE,
-                        r2urp__user=self.user)\
-                .exclude(pk__in=BaseResource.objects
-                         .filter(raccess__immutable=True,
-                                 r2urp__user=self.user,
-                                 r2urp__privilege__lt=PrivilegeCodes.CHANGE))
+                return BaseResource.objects.none()
+
+            if excl:
+                view = BaseResource.objects\
+                    .filter(query)\
+                    .exclude(pk__in=BaseResource.objects
+                             .filter(excl))
+            else:
+                view = BaseResource.objects\
+                    .filter(query)
+
+            # step 2: compute immutable privileges, whose privilege is CHANGE
+            # with no other privilege < CHANGE (OWNER). Since groups can't own,
+            # there is no exclusion for them.
+            uquery = Q(raccess__immutable=True,
+                       r2urp__privilege=PrivilegeCodes.CHANGE,
+                       r2urp__user=self.user)
+            uexcl = Q(raccess__immutable=True,
```

Comment:
The ```uquery``` is already filtering out  resources that are owned by the user. So I thought we don't need ```uexcl``` query. Let me know I can +1 this PR.